_books/ion-1-1/src/macros/macros_by_example.md

```diff
@@ -155,720 +132,160 @@ The outer e-expression receives the results from the expansion of the inner e-ex
-as long as no following parameter is being given an expression. We’ve seen
-this as applied to final `*` parameters, but it also applies to `?`
-parameters:
+In addition to tagged placeholders, Ion 1.1 supports _tagless placeholders_ that specify a
+particular type. These are more restrictive but enable more compact binary encoding.
 
 ```ion
-(macro optionals (a* b? c! d* e? f*)
-  (.make_list a b c d e f))
+(point {x: (:?\int\), y: (:?\int\)})
 ```
 
-Since `d`, `e`, and `f` all accept the empty stream, they can be omitted by invokers. But `c` is required so
-`a` and `b` must always be present, at least as an empty group:
+This macro uses tagless placeholders that require integer arguments. The `\int\` is a type marker
+indicating the placeholder expects an integer value.
 
 ```ion
-(:optionals (::) (::) "value for c") ⇒ ["value for c"]
+(:point 3 17) ⇒ {x: 3, y: 17}
 ```
 
-Now `c` receives the string `"value for c"` while the other parameters are all empty.
-If we want to provide `e`, then we must also provide a group for `d`:
+Tagless placeholders have important restrictions:
+- They cannot accept null values
+- They cannot accept annotated values
+- They cannot be omitted (they're always required)
+- Arguments must match the specified type
 
 ```ion
-(:optionals (::) (::) "value for c" (::) "value for e")
-  ⇒ ["value for c", "value for e"]
+(:point null.int 17)   ⇒ // Error: tagless int does not accept nulls
+(:point a::3 17)       ⇒ // Error: tagless int does not accept annotations
+(:point (:) 17)        ⇒ // Error: tagless parameters cannot be omitted
 ```
 
-## Tagless and fixed-width types
-
-In Ion 1.0, the binary encoding of every value starts off with a “type tag”, an opcode that indicates
-the data-type of the next value and thus the interpretation of the following octets of data. In general,
-these tags also indicate whether the value has annotations, and whether it’s null.
-
-These tags are necessary because the Ion data model allows values of any type to be used
-anywhere. Ion documents are not schema-constrained: nothing forces any part of the data to have a
-specific type or shape. We call Ion “self-describing” precisely because each value
-self-describes its type via a type tag.
-
-If schema constraints are enforced through some mechanism outside the serializer/deserializer,
-the type tags are unnecessary and may add up to a non-trivial amount of wasted space.
-Furthermore, the overhead for each value also includes length information: encoding an
-octet of data takes two octets on the stream.
-
-Ion 1.1 tries to mitigate this overhead in the binary format by allowing macro parameters to use
-more-constrained _tagless types_. These are subtypes of the concrete types,
-constrained such that type tags are not necessary in the binary form. In general this can shave
-4-6 bits off each value, which can add up in aggregate. In the extreme, that octet of data can
-be encoded with no overhead at all.
+## Available Tagless Types
```

Comment:
This sort of stuff is probably going to be repeated in the binary encoding and text encoding chapters, and it's not specific to macros now that we have tagless-element sequences. Let's just link to a `tagless_encodings.md` page. (It can be left as `todo.md` until that page is created.)